app/templates/courses/project-gallery-view.jade

```diff
@@ -1,7 +1,8 @@
 flat-layout
   .container.m-t-3
     p
-      a(v-if="amTeacher", :href="backToClassroomUrl", data-i18n="courses.back_classroom")
+      a(v-if="amSchoolAdministrator", :href="schoolAdministratorBackUrl", data-i18n="courses.back_classroom")
```

Comment:
Using isSchoolAdministrator to decide where we link back to may result in a confusing UX for some users.  For example:

1. I'm a teacher with my own class and I become a school administrator (maybe a department head that also teaches a class or two).
2. I drill down into a class from the My Classes view and arrive at the project gallery
3. I click the back to my classes page which brings me to the school administrator view of the class (read only)

This may be acceptable but we should discuss.  Alternatively we can use some kind of light weight state tracking to determine how we arrived at this page.

Reply:
Alternatively we could say that school administrators cannot be teachers.  This would remove this scenario however I can see this becoming an issue in the future if we ever decide to relax that restriction as I don't expect us to remember something small like this :).

Reply:
Thats a great point. Can we check here if the user is a "schoolAdmin" and "looking at project galleries of his own class or other teacher's class"?

Reply:
Looks like `courseInstanceID` is available in the URL on this page, can we check the user's access to the course instance to determine the back URL.

Reply:
Great points - yes we can absolutely do this. I'll make the change 👍